hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/TestHoodieClientMultiWriter.java

```diff
@@ -516,10 +516,13 @@ public void testMultiWriterWithAsyncTableServicesWithConflict(HoodieTableType ta
       // We want the upsert to go through only after the compaction
       // and cleaning schedule completion. So, waiting on latch here.
       latchCountDownAndWait(scheduleCountDownLatch, 30000);
-      if (tableType == HoodieTableType.MERGE_ON_READ) {
-        // Since the compaction already went in, this upsert has
+      if (tableType == HoodieTableType.MERGE_ON_READ && !(resolutionStrategy instanceof PreferWriterConflictResolutionStrategy)) {
+        // HUDI-6897: Improve SimpleConcurrentFileWritesConflictResolutionStrategy for NB-CC
```

Comment:
This should be immediate followup. Let's fix the constraints in resolution strategy.

Reply:
yeah, that would be a nice improvement.